src/test/java/org/gaul/s3proxy/AwsSdkTest.java

```diff
@@ -300,6 +340,40 @@ public void testAwsV2UrlSigning() throws Exception {
         }
     }
 
+    @Test
+    public void testAwsV2UrlSigningWithOverrideParameters() throws Exception {
+        client = AmazonS3ClientBuilder.standard()
+                .withClientConfiguration(V2_SIGNER_CONFIG)
+                .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+                .withEndpointConfiguration(s3EndpointConfig).build();
+
+        String blobName = "foo";
+        ObjectMetadata metadata = new ObjectMetadata();
+        metadata.setContentLength(BYTE_SOURCE.size());
+        client.putObject(containerName, blobName, BYTE_SOURCE.openStream(),
+                metadata);
+
+        GeneratePresignedUrlRequest generatePresignedUrlRequest =
+                new GeneratePresignedUrlRequest(containerName, blobName);
+        generatePresignedUrlRequest.setMethod(HttpMethod.GET);
+
+        ResponseHeaderOverrides headerOverride = new ResponseHeaderOverrides();
+
+        headerOverride.setContentDisposition("attachment; " + blobName);
+        headerOverride.setContentType("text/plain");
+        generatePresignedUrlRequest.setResponseHeaders(headerOverride);
+
+        Date expiration = new Date(System.currentTimeMillis() +
+                TimeUnit.HOURS.toMillis(1));
+        generatePresignedUrlRequest.setExpiration(expiration);
```

Comment:
testAwsV2SignatureWithOverrideParameters is testing w/ non-presigned URL. testAwsV2UrlSigningWithOverrideParameters is testing w/ presigned URL, which is the same as testAwsV2UrlSigning(). I can add verification for expiration, content dispositioin and type. Or are you insisting to merge this test case to testAwsV2SignatureWithOverrideParameters?